queue/QueueConsumer.js

```diff
@@ -24,13 +52,57 @@ module.exports = class QueueConsumer extends BaseQueue {
     if (!this._channel) {
       throw new Error('Channel not initialized')
     }
+
+    const messageContent = message.content
+
+    const maxMessageRetries = this._maxMessageRetries
+    const deadLetterQueueName = this._deadLetterQueueName
+
+    const { headers } = message.properties
+    const deadLetterHeader = headers[DEAD_LETTER_HEADER]
+
+    let rejectedRecord
+    if (deadLetterHeader) {
+      // find 'rejected' record
+      // TODO: handle other reasons in the future
+      // ex. dead letter messages from TTL expires (not implemented yet)
+      for (const record of deadLetterHeader) {
+        if (record.reason === 'rejected') {
+          rejectedRecord = record
```

Comment:
Just did some googling. There _could_ be more than one record with the reason set to `rejected`, but that's only if more than one queue is set to point rejected messages to the queue's dead letter exchange. In our case, we only have one queue sending rejects to a single exchange (dedicated to that queue), so this won't happen.

Just be safe, I'll traverse the records and find a record where the `queue` matches the consumer's `queueName`.